packages/zoe/contracts/publicAuction.js

```diff
@@ -20,75 +21,90 @@ export const makeContract = harden((zoe, terms) => {
+          closeAuction(zoe, assays, {
+            auctionLogicFn: secondPriceLogic,
+            itemIndex: ITEM_INDEX,
+            bidIndex: BID_INDEX,
+            sellerInviteHandle,
+            allBidHandles,
+          });
+        }
+        return defaultAcceptanceMsg;
+      },
+    });
+    const { invite, inviteHandle } = zoe.makeInvite(seat, {
+      seatDesc: 'bid',
+      auctionedAssets,
+      minimumBid,
+    });
+    return invite;
+  };
 
-      // Save the valid offer
-      creatorOfferHandle = offerHandle;
-      auctionedAssets = payoutRules[0].units;
-      minimumBid = payoutRules[1].units;
-      return defaultAcceptanceMsg;
-    },
-    getMinimumBid: () => minimumBid,
-    getAuctionedAssets: () => auctionedAssets,
-    bid: async escrowReceipt => {
-      const {
-        offerHandle,
-        offerRules: { payoutRules },
-      } = await zoe.burnEscrowReceipt(escrowReceipt);
-
-      // Check that the item is still up for auction
-      const { inactive } = zoe.getStatusFor(harden([creatorOfferHandle]));
-      if (inactive.length > 0) {
-        return rejectOffer(
-          zoe,
-          offerHandle,
-          'The item up for auction has been withdrawn or the auction has completed',
-        );
-      }
-
-      if (allBidHandles.length >= numBidsAllowed) {
-        return rejectOffer(zoe, offerHandle, `No further bids allowed.`);
-      }
-
-      const ruleKinds = ['wantAtLeast', 'offerAtMost'];
-      if (!hasValidPayoutRules(ruleKinds, terms.assays, payoutRules)) {
-        return rejectOffer(zoe, offerHandle);
-      }
+  const makeSellerInvite = () => {
+    const seat = harden({
+      sellAssets: () => {
+        if (
+          auctionedAssets ||
+          !hasValidPayoutRules(['offerAtMost', 'wantAtLeast'], inviteHandle)
+        ) {
+          throw rejectOffer(inviteHandle);
+        }
 
-      if (!isOverMinimumBid(zoe, BID_INDEX, creatorOfferHandle, offerHandle)) {
-        return rejectOffer(zoe, offerHandle, `Bid was under minimum bid`);
-      }
-
-      // Save valid bid and try to close.
-      allBidHandles.push(offerHandle);
-      if (allBidHandles.length >= numBidsAllowed) {
-        closeAuction(zoe, {
-          auctionLogicFn: secondPriceLogic,
-          itemIndex: ITEM_INDEX,
-          bidIndex: BID_INDEX,
-          creatorOfferHandle,
-          allBidHandles,
-        });
-      }
-      return defaultAcceptanceMsg;
-    },
-  });
+        // Save the valid offer
+        sellerInviteHandle = inviteHandle;
+        const { payoutRules } = zoe.getOffer(inviteHandle);
+        auctionedAssets = payoutRules[0].units;
+        minimumBid = payoutRules[1].units;
+        return defaultAcceptanceMsg;
+      },
+    });
+    const { invite, inviteHandle } = zoe.makeInvite(seat, {
+      seatDesc: 'sellAssets',
+    });
+    return invite;
+  };
 
   return harden({
-    instance: publicAuction,
-    assays: terms.assays,
+    invite: makeSellerInvite(),
+    publicAPI: {
```

Comment:
Yeah, good catch. I added a simple check for whether any assets have been put up for auction. 

I'd love to be able to spend more time on redesigning the publicAPI, but unless we can bound it to a few hours of thought, I think we need to try to wrap up this series of changes and release them. Then in the next phase when we are trying to allow for easily adding assays and using subsets of assays in a contract, it might open the design space wide up again and it would be a good time to rethink things like this.